crates/pathfinder/src/rpc/v02/method/get_proof.rs

```diff
@@ -0,0 +1,272 @@
+use anyhow::{anyhow, Context};
+use serde::Deserialize;
+
+use crate::core::{BlockId, ContractAddress, StorageAddress};
+use crate::rpc::v02::RpcContext;
+use crate::state::merkle_tree::ProofNode;
+use crate::state::state_tree::{ContractsStateTree, GlobalStateTree};
+use crate::storage::{ContractsStateTable, StarknetBlocksBlockId, StarknetBlocksTable};
+
+#[derive(Deserialize, Debug, PartialEq, Eq)]
+pub struct GetStorageProofInput {
+    pub contract_address: ContractAddress,
+    pub key: StorageAddress,
+    pub block_id: BlockId,
+}
+
+crate::rpc::error::generate_rpc_error_subset!(
+    GetStorageProofError: ContractNotFound,
+    BlockNotFound
```

Comment:
```suggestion
    GetStorageProofError: BlockNotFound
```
If the contract does not exist, we have to provide a merkle proof indicating that.

Reply:
6012c7cd123b2d15b92e375da7705721658f6e54